src/main/java/uk/gov/hmcts/dts/fact/controllers/admin/AdminLocalAuthoritiesController.java

```diff
@@ -0,0 +1,63 @@
+package uk.gov.hmcts.dts.fact.controllers.admin;
+
+import io.swagger.annotations.ApiOperation;
+import io.swagger.annotations.ApiResponse;
+import io.swagger.annotations.ApiResponses;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.MediaType;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+import org.springframework.web.server.ResponseStatusException;
+import uk.gov.hmcts.dts.fact.config.security.Role;
+import uk.gov.hmcts.dts.fact.model.admin.LocalAuthority;
+import uk.gov.hmcts.dts.fact.services.admin.AdminLocalAuthorityService;
+import uk.gov.hmcts.dts.fact.services.validation.ValidationService;
+
+import java.util.List;
+
+import static org.springframework.http.ResponseEntity.ok;
+import static uk.gov.hmcts.dts.fact.services.admin.AdminRole.FACT_ADMIN;
+import static uk.gov.hmcts.dts.fact.services.admin.AdminRole.FACT_SUPER_ADMIN;
+
+@RestController
+@RequestMapping(
+    path = "/admin/localauthorities",
+    produces = {MediaType.APPLICATION_JSON_VALUE}
+)
+public class AdminLocalAuthoritiesController {
+    private final AdminLocalAuthorityService adminLocalAuthorityService;
+    private final ValidationService validationService;
+
+    @Autowired
+    public AdminLocalAuthoritiesController(AdminLocalAuthorityService localAuthorityService, ValidationService validationService) {
+        this.adminLocalAuthorityService = localAuthorityService;
+        this.validationService = validationService;
+    }
+
+    @GetMapping(path = "/all")
+    @ApiOperation("Return all local authorities")
+    @Role({FACT_ADMIN, FACT_SUPER_ADMIN})
+    public ResponseEntity<List<LocalAuthority>> getAllLocalAuthorities() {
+        return ok(adminLocalAuthorityService.getAllLocalAuthorities());
+    }
+
+    @PutMapping(path = "/{localAuthorityId}")
+    @ApiOperation("Update local authority")
+    @ApiResponses(value = {
+        @ApiResponse(code = 200, message = "Successful", response = LocalAuthority.class),
+        @ApiResponse(code = 400, message = "Invalid Local Authority", response = String.class),
+        @ApiResponse(code = 401, message = "Unauthorized"),
+        @ApiResponse(code = 403, message = "Forbidden"),
+        @ApiResponse(code = 404, message = "Local Authority not found")
+    })
+    @Role({FACT_SUPER_ADMIN})
+    public ResponseEntity<LocalAuthority> updateLocalAuthority(@PathVariable Integer localAuthorityId, @RequestBody LocalAuthority localAuthority) {
+        if (localAuthority != null && validationService.validateLocalAuthority(localAuthority.getName())) {
+            return ok(adminLocalAuthorityService.updateLocalAuthority(localAuthorityId, localAuthority));
+        } else {
+            String error = "Invalid Local Authority: " + ((localAuthority == null) ? "null" : localAuthority.getName());
+            throw new ResponseStatusException(HttpStatus.BAD_REQUEST, error);
```

Comment:
There is nothing wrong with throwing the response status exception directly here. In spring boot sometimes you can have a central exception handler which catches all these exceptions and translate them to relevant response status code. The benefits of doing that is you will have a common location for these various exception vs response status. I prefer this way normally so added the class in 'GlobalControllerExceptionHandler' to catch all these exceptions thrown from the controllers. This class is like a controller gate which catches all exceptions before they get out of the API. You can look at invalid postcode example from 'InvalidPostcodeException' class and 'invalidPostcodeExceptionHandler' method. For the postcode validation, we return a list of invalid postcodes. Not saying you must do the same but just telling you the difference.

Reply:
The second thing is we don't need an else statement here because the previous if statement is returning from the method. You can throw the exception directly after the if statement. This is going to general a sonarqube code smell

Reply:
Something to think about...do we need to pass in and return a local authority object, or is that better just to pass in and return a string. You can already get the ID from the path

Reply:
I agree with all of these points. Code updated accordingly.